libpromises/locks.c

```diff
@@ -587,6 +614,12 @@ static bool KillLockHolder(const char *lock)
 
     CloseLock(dbp);
 
+    if (!IsCfengineProcess(lock_data.pid)) {
+        Log(LOG_LEVEL_VERBOSE,
+            "Lock holder with pid %d was replaced by a non cfengine process, do not try to kill it!, ", lock_data.pid);
+        return true;
```

Comment:
So the function returns `true` here in this case, but it doesn't really kill the lock holder. That can be a source of issues.

Reply:
This is still applicable.